experiments/conductor/cmd/runner/mock_commands.go

```diff
@@ -160,25 +178,181 @@ func readScriptYaml(opts *RunnerOptions, branch Branch) {
 		return
 	}
 
+	// Read and parse the script file
 	data, err := os.ReadFile(scriptFullPath)
 	if err != nil {
-		fmt.Println("Error reading file:", err)
+		log.Printf("Error reading file %s: %v", scriptFullPath, err)
 		return
 	}
-	dataStr := string(data)
-	if !strings.Contains(dataStr, " create") {
-		fmt.Println("WARNING: Script doesn't contain a CREATE command")
+
+	// Parse YAML content
+	var scriptContent []map[string]string
+	err = yaml.Unmarshal(data, &scriptContent)
+	if err != nil {
+		log.Printf("Error parsing YAML from %s: %v", scriptFullPath, err)
+		return
 	}
-	if !strings.Contains(dataStr, " update") {
-		fmt.Println("WARNING: Script doesn't contain an UPDATE command")
+
+	// Check commands and collect warnings
+	var warnings []string
+	hasCreate := false
+	hasDelete := false
+	hasList := false
+
+	for _, line := range scriptContent {
+		if cmd, ok := line["exec"]; ok && strings.Contains(cmd, "gcloud") {
+			if strings.Contains(cmd, " create ") {
+				hasCreate = true
+			}
+			if strings.Contains(cmd, " delete ") {
+				hasDelete = true
+			}
+			if strings.Contains(cmd, " list ") {
+				hasList = true
+			}
+		}
 	}
-	if !strings.Contains(dataStr, " delete") {
-		fmt.Println("WARNING: Script doesn't contain a DELETE command")
+
+	if !hasCreate {
+		warning := "No Create found"
+		warnings = append(warnings, warning)
+		log.Printf("WARNING: %s in %s", warning, scriptFile)
+	}
+	if !hasDelete {
+		warning := "No Delete found"
+		warnings = append(warnings, warning)
+		log.Printf("WARNING: %s in %s", warning, scriptFile)
 	}
-	if strings.Contains(dataStr, " list") {
-		fmt.Println("WARNING: Script contains a LIST command")
+	if hasList {
+		warning := "List found"
+		warnings = append(warnings, warning)
+		log.Printf("WARNING: %s in %s", warning, scriptFile)
 	}
-	fmt.Println(dataStr)
+
+	// Read existing all_scripts.yaml or create new ScriptData
```

Comment:
I think when I read, I only want the latest information based on the list of metadata entries. I do not want to read information other than the entries I provided in the branch yaml. Otherwise I'd add those entries. So I think overriding the entire file may be a more intuitive option here?

Reply:
iam thinking with the --for-resource, this may not be true anymore ? 
Because we filter in only those resources. Iam ok to change this. May be we will try these out and update afterwards.